js/apiCalls.mjs

```diff
@@ -0,0 +1,83 @@
+import { local } from "./global.mjs";
+
+export const api = {
+    
+    'baseUrl' : 'https://v2.api.noroff.dev',
+    'logInEndPoint' : '/auth/login',
+    'createEndpoint'  : '/auth/register',
+    'getApiKeyEndpoint' : '/auth/create-api-key',
+    'postsEndpoint' : '/social/posts',
+    'profileEndPoint' : '/social/profiles',
+    'searchEndpoint' : '/social/posts/search?q=',
+    'getApi' : 'GET',
+    'postApi' : 'POST',
+    'deleteApi' : 'DELETE',
+    'putApi' : 'PUT',
+    /**
+    * Make an api call
+    * @param {any} data String for search, number to target post, object to post forms
+    * @param {string} endPoint use the variable that gives the correct endpoint (located in apiCalls.js)
+    * @param {string} method the Method needed
+    * @param {array} endUrl url string to add
+    * @returns {response}
+    */
+    async call(data,endPoint,method,endUrl) {
```

Comment:
Please run a formatting tool like prettier on all JS files. You can install this as a dev dependency `npm i -D prettier` and set it up as a format task in package.json to do this automatically.